interfaces/cython/cantera/test/test_utils.py

```diff
@@ -0,0 +1,117 @@
+
+    def test_none(self):
+        out, held_type = _py_to_any_to_py(None)
+        self.assertEqual(out, None)
+        self.assertEqual(held_type, "void")
+
+    def test_set(self):
+        # Sets are converted to lists
+        self.check_inexact_conversion({'a', 'b'}, 'vector<string>')
+
+    def test_empty_list(self):
+        self.check_conversion([])
+
+    def test_empty_ndarray(self):
+        self.check_inexact_conversion(np.ndarray((0,)))
+
+    def test_empty_dict(self):
+        self.check_conversion({})
+
+    def test_scalar_string(self):
+        self.check_conversion('spam', 'string')
+
+    def test_scalar_int(self):
+        self.check_conversion(3, 'long int')
+
+    def test_scalar_float(self):
+        self.check_conversion(3.1415, 'double')
+
+    def test_scalar_bool(self):
+        self.check_conversion(True, 'bool')
+
+    def test_list_string(self):
+        self.check_conversion(['spam', 'eggs'], 'vector<string>')
+
+    def test_list_int(self):
+        self.check_conversion([1, 2, 3], 'vector<long int>')
+
+    def test_list_float(self):
+        self.check_conversion([1., 2., 3.], 'vector<double>')
+
+    def test_list_bool(self):
+        self.check_conversion([True, False], 'vector<bool>')
+
+    def test_list_various(self):
+        self.check_conversion([True, 'spam', 3, 4., {'foo': 'bar'}],
+                              'vector<AnyValue>')
+
+    def test_tuple(self):
+        self.check_inexact_conversion((True, 'spam', 3, 4.), 'vector<AnyValue>')
+
+    def test_ndarray1(self):
+        self.check_inexact_conversion(np.random.randn(10), 'vector<double>')
+
+    def test_ndarray2(self):
+        self.check_inexact_conversion(np.random.randn(3, 2), 'vector<vector<double>>')
+
+    def test_ndarray3(self):
+        # Each inner AnyValue holds a vector<vector<double>>
+        self.check_inexact_conversion(np.random.randn(3, 2, 4), 'vector<AnyValue>')
+
+    def test_nested_string(self):
+        self.check_conversion([['spam', 'eggs'], ['foo', 'bar']],
+                              'vector<vector<string>>')
+
+    def test_nested_int(self):
+        self.check_conversion([[1, 2, 3], [4, 5, 6]], 'vector<vector<long int>>')
+
+    def test_nested_float(self):
+        self.check_conversion([[1., 2., 3.], [4., 5., 6.]], 'vector<vector<double>>')
+
+    def test_nested_bool(self):
+        self.check_conversion([[True, False], [False, True]], 'vector<vector<bool>>')
+
+    def test_multi_dict(self):
+        vv = [{'a': [['spam', 'eggs'], ['foo', 'bar']], 'b': {'c': 4}}, {'d': 3}]
+        self.check_conversion(vv, 'vector<AnyMap>')
+
+    def test_dict(self):
+        self.check_conversion({'a': 1, 'b': 2., 'c': 'eggs', 'd': True}, 'AnyMap')
+
+    def test_nested_dict(self):
+        self.check_conversion({'a': 1, 'b': 2., 'c': {'d': 'eggs'}}, 'AnyMap')
+
+    def test_unconvertible(self):
+        class Foo: pass
+        self.check_raises(Foo(), ct.CanteraError, "Unable to convert")
```

Comment:
You can eliminate this class declaration by instantiating `object()` in the `check_raises()` call.